daps-xslt/mallard/docbook.xsl

```diff
@@ -175,7 +163,8 @@
 
   <xsl:template match="book/article | d:book/d:article">
     <xsl:param name="node" select="."/>
-    <link href="help:{$packagename}/{@id}">
+    <xsl:variable name="id" select="($node/@id|$node/@xml:id)[1]"/>
```

Comment:
There might be no result at all if there is no `id` attribute. Not sure if this is necessary, but you could extend it as follows:

```xml
<xsl:variable name="id">
   <xsl:choose>
      <xsl:when test="($node/@id|$node/@xml:id)[1]">
         <xsl:value-of select="($node/@id|$node/@xml:id)[1]"/>
      </xsl:when>
      <xsl:otherwise>
         <xsl:value-of select="generate-id($node)"/>
      </xsl:otherwise>
   </xsl:choose>
</xsl:variable>
```

Same applies for other variables with the same structure.